code/modules/mob/living/basic/pets/cat/cat_ai.dm

```diff
@@ -0,0 +1,331 @@
+		if(SPT_PROB(leave_home_chance, seconds_per_tick))
+			controller.set_blackboard_key(BB_CAT_HOME, living_pawn.loc)
+		return SUBTREE_RETURN_FINISH_PLANNING
+
+	if(SPT_PROB(reside_chance, seconds_per_tick))
+		controller.queue_behavior(/datum/ai_behavior/find_and_set/valid_home, BB_CAT_HOME, /obj/structure/cat_house)
+
+/datum/ai_behavior/find_and_set/valid_home/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	for(var/obj/structure/cat_house/home in oview(search_range, controller.pawn))
+		if(home.resident_cat)
+			continue
+		return home
+
+	return null
+
+/datum/ai_behavior/enter_cat_home
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION | AI_BEHAVIOR_REQUIRE_REACH
+
+/datum/ai_behavior/enter_cat_home/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/enter_cat_home/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/structure/cat_house/home = controller.blackboard[target_key]
+	var/mob/living/basic/living_pawn = controller.pawn
+	if(living_pawn == home.resident_cat || isnull(home.resident_cat))
+		living_pawn.melee_attack(home)
+		finish_action(controller, TRUE, target_key)
+		return
+
+	finish_action(controller, FALSE, target_key)
+
+/datum/ai_behavior/enter_cat_home/finish_action(datum/ai_controller/controller, success, target_key)
+	. = ..()
+	controller.clear_blackboard_key(target_key)
+
+
+/datum/ai_planning_subtree/flee_target/from_flee_key/cat_struggle
+	flee_behaviour = /datum/ai_behavior/run_away_from_target/cat_struggle
+
+/datum/ai_behavior/run_away_from_target/cat_struggle
+	clear_failed_targets = TRUE
+
+/datum/ai_planning_subtree/territorial_struggle
+	///chance we become hostile to another cat
+	var/hostility_chance = 5
+
+/datum/ai_planning_subtree/territorial_struggle/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/mob/living/living_pawn = controller.pawn
+	if(living_pawn.gender != MALE || !SPT_PROB(hostility_chance, seconds_per_tick))
+		return
+	if(controller.blackboard_key_exists(BB_TRESSPASSER_TARGET))
+		controller.queue_behavior(/datum/ai_behavior/territorial_struggle, BB_TRESSPASSER_TARGET, BB_HOSTILE_MEOWS)
+		return SUBTREE_RETURN_FINISH_PLANNING
+
+	controller.queue_behavior(/datum/ai_behavior/find_and_set/cat_tresspasser, BB_TRESSPASSER_TARGET, /mob/living/basic/pet/cat)
+
+/datum/ai_behavior/find_and_set/cat_tresspasser/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	var/list/ignore_types = controller.blackboard[BB_BABIES_CHILD_TYPES]
+	for(var/mob/living/basic/pet/cat/potential_enemy in oview(search_range, controller.pawn))
+		if(potential_enemy.gender != MALE)
+			continue
+		if(is_type_in_list(potential_enemy, ignore_types))
+			continue
+		if(isnull(potential_enemy.ai_controller))
```

Comment:
cache `potential_enemy.ai_controller` to a variable named something like `enemy_ai_controller` right before this step so it's easier to read.